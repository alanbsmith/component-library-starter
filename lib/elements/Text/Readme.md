
##### DESCRIPTION
This is our basic text element. It's styled exactly like P, but is intended for single words or short phrases. It has four sizes: extraSmall, small, medium (default), and large. It also has a four font weights available: light, normal (default), medium, and bold. You can use it on its own, or extend the styles in a block.


```jsx
<div>
  <h5>EXTRA SMALL</h5>
  <Text modifiers={['extraSmall']}>
    Don't forget to read the fine print!
  </Text>

  <h5>SMALL</h5>
  <Text modifiers={['small']}>
    a helpful caption
  </Text>

  <h5>DEFAULT</h5>
  <Text>
    A short description
  </Text>

  <h5>LARGE</h5>
  <Text modifiers={['large']}>
    Sign up here!
  </Text>
</div>
```
