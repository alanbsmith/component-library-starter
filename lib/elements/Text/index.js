import styled from "styled-components";
import { applyStyleModifiers } from "styled-components-modifiers";
import { rem } from 'polished';

import {
  fontColors,
  fontSizes,
  fontStyles,
  fontWeights
} from "../../modifiers";

const modifierConfig = {
  ...fontColors,
  ...fontSizes,
  ...fontStyles,
  ...fontWeights,
};

const Text = styled.span`
  font-size: ${props => rem(props.theme.fontSizes.medium)};
  margin: 0;
  ${applyStyleModifiers(modifierConfig)};
`;

Text.modifiers = modifierConfig;

/** @component */
export default Text;
