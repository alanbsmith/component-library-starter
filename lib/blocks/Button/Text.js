import styled from 'styled-components';

import { Text as BaseText } from '../../elements';

const Text = styled(BaseText)`
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;
`;

export default Text;
