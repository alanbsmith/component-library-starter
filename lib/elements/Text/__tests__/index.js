import React from 'react';
import Text from '../index';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';

describe('Text Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Text>Some Text</Text>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the italic modifier', () => {
    const tree = renderWithTheme(
      <Text modifiers={['italic']}>Some Text</Text>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the extraSmall modifier', () => {
    const tree = renderWithTheme(
      <Text modifiers={['extraSmall']}>Some Text</Text>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the small modifier', () => {
    const tree = renderWithTheme(
      <Text modifiers={['small']}>Some Text</Text>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the large modifier', () => {
    const tree = renderWithTheme(
      <Text modifiers={['large']}>Some Text</Text>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the fontWeightBold modifier', () => {
    const tree = renderWithTheme(
      <Text modifiers={['fontWeightBold']}>Some Text</Text>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
