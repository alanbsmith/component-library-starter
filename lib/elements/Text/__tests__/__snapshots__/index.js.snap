// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Element adds the extraSmall modifier 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
  font-size: 0.75rem;
}

<span
  className="c0"
>
  Some Text
</span>
`;

exports[`Text Element adds the fontWeightBold modifier 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
}

<span
  className="c0"
>
  Some Text
</span>
`;

exports[`Text Element adds the italic modifier 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
}

<span
  className="c0"
>
  Some Text
</span>
`;

exports[`Text Element adds the large modifier 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
  font-size: 1.125rem;
}

<span
  className="c0"
>
  Some Text
</span>
`;

exports[`Text Element adds the small modifier 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
  font-size: 0.875rem;
}

<span
  className="c0"
>
  Some Text
</span>
`;

exports[`Text Element renders correctly 1`] = `
.c0 {
  font-size: 1rem;
  margin: 0;
}

<span
  className="c0"
>
  Some Text
</span>
`;
